pkg/plugin/connector/builtin/v1/internal/toplugin/destination.go

```diff
@@ -23,7 +25,7 @@ func DestinationConfigureRequest(in map[string]string) cpluginv1.DestinationConf
 	out := cpluginv1.DestinationConfigureRequest{}
 	if len(in) > 0 {
 		// gRPC sends `nil` if the map is empty, match behavior
-		out.Config = in
+		out.Config = maps.Clone(in)
```

Comment:
It's interesting that we're cloning it. Why so?

Reply:
In builtin plugins we were passing the very same map (pointer) that was used to store the config in the `Processor` and `Connector` entities. So if the plugin changes that map, we will see the change. Now we clone it so the plugin can do whatever it wants with that map without side effects.

Reply:
Btw, why the change now? It showed up as a race condition in tests. Apparently a change in the SDK now causes the map to be manipulated.